### PR TITLE
Add a "How to start a new service" branch with sample commits and code

### PR DESCRIPTION
**Pull request by dhermes**
@jonparrott and @waprin have both suggested something like this and my discussion with @mckoss today pushed me over the "OK Jon and Bill" edge.

I am not quite done yet, but wanted to get it in the hands of those "in the know" now, so you can give ideas and feedback as it becomes a living thing.

- The commit content and commit messages are actually important, so please review them
- Should this go in it's own branch (referred to from CONTRIBUTING.rst) or should we check it in to `master` (checking in to master will help keep it from going stale, but may make the commits harder to find)
- Is there an actual fake service we can talk to? @bjwatson do you know

PS: H/T to http://unicorn.namegeneratorfun.com/

### CODE REVIEW

**Comment by jonparrott** [comment]
> Is there an actual fake service we can talk to? 

You could always write an emulator for your fake service (in both HTTP/JSON and gRPC).

**Comment by jonparrott** [comment]
> Should this go in it's own branch (referred to from CONTRIBUTING.rst) or should we check it in to master (checking in to master will help keep it from going stale, but may make the commits harder to find)

My initial thought is to check it into master. The commits themselves are less important than the content here, and I'm concerned not checking it in will make it drift out of date quickly.

**Comment by waprin** [comment]
Looks really helpful!

Makes sense to go into master, makes less sense to me to be at a top level, rather than maybe in some subdirectory. Maybe demonstrate system tests?

**Comment by daspecster** [comment]
Ditto what @waprin said. I don't think some other branch makes a lot of sense.
There should definitely be a reference to it in `CONTRIBUTING.rst` as well.

File: shiny/setup.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    'author': 'Google Cloud Platform',
+    'author_email': 'jjg+google-cloud-python@google.com',
+    'scripts': [],
+    'url': 'https://github.com/GoogleCloudPlatform/google-cloud-python',
+    'license': 'Apache 2.0',
+    'platforms': 'Posix; MacOS X; Windows',
+    'include_package_data': True,
+    'zip_safe': False,
+    'classifiers': [
+        'Development Status :: 4 - Beta',
```
Comment (bjwatson) [inline shiny/setup.py]: Should probably be Alpha for a new API.

**Comment by dhermes** [comment]
FYI: https://github.com/dhermes/google-shiny-service

**Comment by bjwatson** [comment]
@dhermes Is "Shiny New Service" something that the user would deploy in their own GCP project?

**Comment by bjwatson** [comment]
BTW, my wife and I enjoyed the unicorn name picker link. :)

**Comment by mckoss** [comment]
Should there be Sphinx docs generation templates included?

**Comment by dhermes** [comment]
@mckoss Yes and no. Yes, our sample package should show people how docs work. No, if I checked in `rst` files, this fake service would end up in the real docs.

But, I've been leaning this way for a while and recent discussions with @mckoss have solidified it: (I think) the current approach is unsustainable at scale and we should switch to using `sphinx-apidoc` as [`oauth2client`][2] and [`google-auth-library-python`][1] (due to @jonparrott) do.

[1]: https://github.com/GoogleCloudPlatform/google-auth-library-python/blob/cadec899657213dfff65299567225df6e2ba239f/tox.ini#L62
[2]: https://github.com/google/oauth2client/blob/92e4ad332992b91d4674335f3d83a7181e38967d/scripts/build_docs.sh#L22

**Comment by lukesneeringer** [comment]
@dhermes Is this still a thing / in progress? Or is it dead? If it is dead for all intents and purposes, we should close it.

**Comment by dhermes** [comment]
@lukesneeringer I would like for it to be a thing, but we had a previous discussion where you indicated that it shouldn't be a thing?

**Comment by lukesneeringer** [comment]
I might not be understanding the context of what this is, then? (I recall no such discussion.)

**Comment by lukesneeringer** [comment]
This issue may no longer be relevant due to its age. Feel free to re-open.

**Comment by dhermes** [comment]
@lukesneeringer You and I have had two or three conversations about the value (or lack of value) for such a "fake package" to exist.

I'm happy to "update" it if you think it's worthwhile, but will leave it alone otherwise.

We **previously** had 4 or 5 not so great experiences with other teams doing their own implementations and wanted a boilerplate we could point them at.

**Pull request closed without merging**